pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/EqualsPredicateEvaluatorFactory.java

```diff
@@ -54,30 +57,36 @@ public static BaseDictionaryBasedPredicateEvaluator newDictionaryBasedEvaluator(
    */
   public static BaseRawValueBasedPredicateEvaluator newRawValueBasedEvaluator(EqPredicate eqPredicate,
       DataType dataType) {
+    String value = eqPredicate.getValue();
     switch (dataType) {
       case INT:
-        return new IntRawValueBasedEqPredicateEvaluator(eqPredicate);
+        return new IntRawValueBasedEqPredicateEvaluator(Integer.parseInt(value));
```

Comment:
Are these changes related to BOOLEAN and TIMESTAMP support?

Reply:
Yes, because `BOOLEAN` and `TIMESTAMP` is stored as another type, making these changes so that the code can be reused.